src/internal/system/partitions.hpp

```diff
@@ -31,23 +32,27 @@ namespace srf::internal::system {
 class System;
 class Topology;  // IWYU pragma: keep
 
+// TODO(unknown): flatten partition here
+
 class Partitions
 {
   public:
     Partitions(const Topology& topology, const Options& options);
     Partitions(const System& system);
 
+    const std::vector<Partition>& flattened() const;
```

Comment:
Why `flattened()`? Why not just `partitions()` or `all_partitions()`?

Reply:
There is a comment explaining this on one of the files. 

The host and device partition are hierarchical where there is a possibility, depending on option where more than one device shared the same host partition, so those host resources are shared. 

However, each device defines a partition since it will have its own unique network and device resources. 

Just we flatten the partitions such that it's a flat list where the first N partitions correspond to N cuda devices. If there are more than N partition, those partitions do not have devices attached. 

The flattened partitions provide the view of what's available to a given GPU. 

The plan is to exposes the set of flattened partitions and their resources to the builder and the runtime respectively.

Reply:
Can we put this in a comment on that property?

Reply:
Done